Cabal/src/Distribution/System.hs

```diff
@@ -164,7 +164,7 @@ buildOS = classifyOS Permissive System.Info.os
 --
 data Arch = I386  | X86_64  | PPC  | PPC64 | Sparc
           | Arm   | AArch64 | Mips | SH
-          | IA64  | S390
+          | IA64  | S390    | S390X
```

Comment:
this is a breaking change. You cannot "backport" it.

(it also subtly changes cabal format specification).